sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -676,10 +676,9 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
   // Confirm at least one memory is available in the device
   assert(ZeAvailMemCount > 0);
 
-  ze_device_memory_properties_t *ZeDeviceMemoryProperties;
+  std::vector<ze_device_memory_properties_t> ZeDeviceMemoryProperties;
   try {
-    ZeDeviceMemoryProperties =
-        new ze_device_memory_properties_t[ZeAvailMemCount]();
+    ZeDeviceMemoryProperties.reserve(ZeAvailMemCount);
```

Comment:
Just saw this, but I think you should replace this with `resize`. `reserve` only allocates storage and does not change the number of elements. `resize` will change the size to `ZeAvailMemCount` and default-insert the elements.

Reply:
Makes sense. Opened a PR: https://github.com/intel/llvm/pull/2126